liquidator/liquidator.js

```diff
@@ -248,22 +268,105 @@ class Liquidator {
-      try {
-        await liquidation.call({ from: this.account });
-      } catch (error) {
-        this.logger.error({
+      const notEnoughCollateral = await liquidation
+        .call({ from: this.account })
+        .then(() => false)
+        .catch(() => true);
+
+      // Attempts to swap some capital for tokenCurrency
+      // Mutates the state of notEnoughCollateral in this logical scope
+      if (notEnoughCollateral) {
+        const tokenCurrency = await this.empClient.getTokenCurrency();
+
+        this.logger.info({
           at: "Liquidator",
-          message:
-            "Failed to liquidate position: not enough synthetic (or large enough approval) to initiate liquidation❌",
-          sponsor: position.sponsor,
-          inputPrice: scaledPrice.toString(),
-          position: position,
-          minLiquidationPrice: this.liquidationMinPrice,
-          maxLiquidationPrice: maxCollateralPerToken.toString(),
-          tokensToLiquidate: tokensToLiquidate.toString(),
-          error
+          message: `Attempting to convert reserve currency ${this.reserveCurrencyAddress} to tokenCurrency ${tokenCurrency.address} ❌`
         });
-        continue;
+
+        // Reverse calculation to estimate how much capital we need to get the amount of `tokensToLiquidate`
+        const reserveWeiNeeded = await this.oneInch.getExpectedReturn({
+          fromToken: tokenCurrency.address,
+          toToken: this.reserveCurrencyAddress,
+          amountWei: tokensToLiquidate.toString()
+        });
+
+        // This is used as a reference point to determine slippage
+        const oneGweiReturn = await this.oneInch.getExpectedReturn({
+          fromToken: tokenCurrency.address,
+          toToken: this.reserveCurrencyAddress,
+          amountWei: this.toWei("1", "gwei")
+        });
+
+        const reserveWeiNeededBN = this.toBN(reserveWeiNeeded);
+        const oneGweiReturnBN = this.toBN(oneGweiReturn);
+        const tokensToLiquidateBN = this.toBN(tokensToLiquidate);
+        const oneGweiBN = this.toBN(this.toWei("1", "gwei"));
+        const oneWeiBN = this.toBN(this.toWei("1"));
+
+        // Slippage = (idealReserveWeiNeeded - reserveWeiNeeded) / idealReserveWeiNeeded
+        // Where idealReserverWei = oneGweiReturn * tokensToLiquidate / 1 gwei
+        const idealReserveWeiNeededBN = oneGweiReturnBN.mul(tokensToLiquidateBN).div(oneGweiBN);
+
+        let slippage;
+        // IdealReserveWeiNeededBN is 0 if it isn't on exchanges yet
+        // Can't div by 0 so doing this
+        if (idealReserveWeiNeededBN.eq(this.toBN("0"))) {
+          slippage = "0";
+        } else {
+          slippage = this.fromWei(
+            idealReserveWeiNeededBN
+              .sub(reserveWeiNeededBN)
+              .mul(oneWeiBN)
+              .div(idealReserveWeiNeededBN)
+          );
+        }
+
+        // Don't allow slippage of more than 10%
+        if (parseFloat(slippage) > 0.1) {
```

Comment:
We should make the slippage threshold a state variable that can be updated via the construction config object

Reply:
yup, agreed. Ideally along with the reserver currency; they should both be configs that have defaults and can be overridden the same way all other configs are dealt with.

Reply:
@nicholaspai @chrismaree Totally agreed, but I would like to address the [current concerns I have with the current config](https://github.com/UMAprotocol/protocol/pull/1855#discussion_r472529578) before adding in more parameters

Reply:
I mentioned above, but if you try to tackle the changes to the current config in this PR this PR will take much longer to merge.